Sources/XiEditor/AppDelegate.swift

```diff
@@ -285,6 +264,38 @@ class AppDelegate: NSObject, NSApplicationDelegate {
     }
 
     //- MARK: - helpers
+    func setupConfigDirectory() {
+        let applicationDirectory = FileManager.default.urls(
+            for: .applicationSupportDirectory,
+            in: .userDomainMask)
+            .first!
+            .appendingPathComponent("XiEditor")
+        let preferencesDirectory = applicationDirectory.appendingPathComponent(PREFERENCES_FILE_NAME)
+
+        // Create application support directory and copy preferences file first, then
+        // copies the example config to `preferences.xiconfig` if it isn't found.
+        if !FileManager.default.fileExists(atPath: applicationDirectory.path) {
+            do {
+                try FileManager.default.createDirectory(at: applicationDirectory,
+                                                        withIntermediateDirectories: true,
+                                                        attributes: nil)
+                let preferencesPath = applicationDirectory.appendingPathComponent(PREFERENCES_FILE_NAME)
+                let defaultConfigPath = Bundle.main.url(forResource: "client_example", withExtension: "toml")
+                try FileManager.default.copyItem(at: defaultConfigPath!, to: preferencesPath)
+            } catch let err  {
+                fatalError("Failed to create application support directory \(applicationDirectory.path). \(err)")
```

Comment:
I'm not sure why this error should be fatal and the other one shouldn't. I don't know if there are some circumstances (sandboxing?) where we ever expect to fail here, but if we do fail showing a message to the user is better than hard crashing.